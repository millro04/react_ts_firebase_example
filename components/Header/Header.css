.header-wrapper {
    text-align: center;
  }
  
.header-text {
    font-size: 36px;
  }
  