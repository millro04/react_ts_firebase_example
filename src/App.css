.App {
  text-align: center;
}

.navbar-wrapper {
  padding-top: 20px;
}