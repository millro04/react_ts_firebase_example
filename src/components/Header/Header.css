.header-wrapper {
    text-align: center;
    padding-top: 25px;
  }
  
.header-text {
    font-size: 36px;
  }

  .header-description {
    padding-top: 10px;
  }