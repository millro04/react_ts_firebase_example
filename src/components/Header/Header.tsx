import React from 'react';
import './Header.css';

const Header = () => {
  return (
    <div className="header-wrapper">
      <header className="header-text">
          Reading Tracker
      </header>
    </div>
  );
}

export default Header;
