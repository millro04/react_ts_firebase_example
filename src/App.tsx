import React from 'react';
import './App.css';
import Header from './components/Header/Header';
import DataEntry from './components/DataEntry/DataEntry';

export default class App extends React.Component {
  render() {
    return (
      <div className="App">
        <Header />
        <DataEntry />
      </div>
    );
  }
}