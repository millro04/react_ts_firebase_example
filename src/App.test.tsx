import React from 'react';
import App from './App';
import Header from './components/Header/Header';
import { shallow, ShallowWrapper, configure, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';
configure({adapter: new Adapter()});


test('App component renders', () => {
  const component: ShallowWrapper = shallow(<App />);
  expect(component).toMatchSnapshot();
});


test('App Component renders Header Component', () => {
  const app: ShallowWrapper = shallow(<App />);
  expect(app.find(Header).length).toBeTruthy();
});

test('Renders the DataEntry tab initially', () => {
  const wrapper = mount(
    <Router><App /></Router>
  );
  expect(location.pathname).toEqual('/dataentry');
});


test('Renders the Dashboard when we click on the tab', () => {
  const wrapper = mount(
    <Router><App /></Router>
  );
  expect(location.pathname).toEqual('/dataentry');

  const link = wrapper.find({ className: 'nav-tab' }).at(1);
  link.simulate('click', { button: 0 });

  expect(location.pathname).toEqual('/dashboard')
});
