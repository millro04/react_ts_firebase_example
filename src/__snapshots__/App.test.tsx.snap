// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders 1`] = `
<div
  className="App"
>
  <div
    className="header-wrapper"
  >
    <header
      className="header-text"
    >
      Reading Tracker
    </header>
  </div>
</div>
`;
